libs/scrap/src/wayland/pipewire.rs

```diff
@@ -241,15 +251,16 @@ impl Recorder for PipeWireRecorder {
         if self.buffer.is_none() {
             return Err(Box::new(GStreamerError("No buffer available!".into())));
         }
-        Ok(PixelProvider::BGR0(
-            self.width,
-            self.height,
-            if self.is_cropped {
-                self.buffer_cropped.as_slice()
-            } else {
-                self.buffer.as_ref().unwrap().as_slice()
-            },
-        ))
+        let buf = if self.is_cropped {
+            self.buffer_cropped.as_slice()
+        } else {
+            self.buffer.as_ref().unwrap().as_slice()
+        };
+        match self.pix_fmt.as_str() {
+            "BGRx" => Ok(PixelProvider::BGR0(self.width, self.height, buf)),
+            "RGBx" => Ok(PixelProvider::RGB0(self.width, self.height, buf)),
+            _ => unreachable!(),
```

Comment:
Remove `unreachable!()` to avoid potential problems in the future.
```rust
            _ => {
                Err(Box::new(GStreamerError(format!("Unreachable! Unknown pix_fmt, {}", &self.pix_fmt))))
            },
```

Reply:
Thank you for reviewing!